File: code/backend/foodometer-item-service/src/main/java/com/springmicro/foodometer/exception/ExceptionResponse.java
package com.springmicro.foodometer.exception;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ExceptionResponse {
    private String message;
    private LocalDateTime localDateTime;
    private List<String> stackTrace;
}

File: code/frontend/food-o-gui/src/app/_model/exception-response.ts
export class ExceptionResponse {
    message: string;
    localDateTime: Date;
    stackTrace: string[];

    constructor(message?: string,
        localDateTime?: Date,
        stackTrace?: string[]) {
        this.message = message;
        this.localDateTime = localDateTime;
        this.stackTrace = stackTrace;
    }
}
